executorCommon/src/main/scala/dx/executor/WorkflowExecutor.scala

```diff
@@ -108,16 +108,34 @@ abstract class WorkflowExecutor[B <: Block[B]](jobMeta: JobMeta) {
   }
 
   /**
-    * Evaluates any expressions in workflow outputs.
+    * Evaluates any expressions in workflow outputs. The job output folder is set
+    * to the job name, which is the concatenation of `name` and `nameDetail`.
+    * @param executableLink the executable to launch
+    * @param name the job name
+    * @param inputs the job inputs
+    * @param nameDetail: a suffix to add to the job name
+    * @param instanceType the instance type to use for the new job
+    * @param folder optional output folder
     */
   protected def launchJob(executableLink: ExecutableLink,
                           name: String,
                           inputs: Map[String, (Type, Value)],
                           nameDetail: Option[String] = None,
-                          instanceType: Option[String] = None): (DxExecution, String) = {
+                          instanceType: Option[String] = None,
+                          folder: Option[String] = None): (DxExecution, String) = {
     val jobName: String = nameDetail.map(hint => s"${name} ${hint}").getOrElse(name)
+    val outputFolder = if (separateOutputs) {
```

Comment:
Cannot jobName be non-unique for a set of calls within an analysis?

Reply:
It will always be unique within a given scope, and folders are nested by scope, so there should never be a folder name collision.